src/newrelic/processor.go

```diff
@@ -456,6 +462,49 @@ func (p *Processor) processConnectAttempt(rep ConnectAttempt) {
 		NewHarvest(time.Now(), app.connectReply.EventHarvestConfig.EventConfigs), p.processorHarvestChan)
 }
 
+func processLogEventLimits(app *App) {
+
+	if nil == app {
+		log.Warnf("processLogEventLimits() called with *App == nil")
+		return
+	}
+
+	if nil == app.info {
+		log.Warnf("processLogEventLimits() called with app.info == nil")
+		return
+	}
+
+	if nil == app.connectReply {
+		log.Warnf("processLogEventLimits() called with  app.connectReply == nil")
+		return
+	}
+
+	// need to compare agent limits to limits returned from the collector
+	// and choose the smallest value
+	agentLogLimit := app.info.AgentEventLimits.LogEventConfig.Limit
+	agentReportPeriod := float64(limits.DefaultReportPeriod)
+
+	collectorLogLimit := app.connectReply.EventHarvestConfig.EventConfigs.LogEventConfig.Limit
+	collectorReportPeriod := float64(app.connectReply.EventHarvestConfig.EventConfigs.LogEventConfig.ReportPeriod)
+
+	// convert agent log limit to sampling period used for log events
+	agentLogLimit = int((float64(agentLogLimit) * collectorReportPeriod) / agentReportPeriod)
+
+	log.Debugf("handling log limits: agent_report_period = %f collector_report_period = %f", agentReportPeriod, collectorReportPeriod)
+	log.Debugf("handling log limits: agent_log_limit = %d agentLogLimit = %d", agentLogLimit, collectorLogLimit)
+
+	finalLogLimit := collectorLogLimit
+	if agentLogLimit < collectorLogLimit {
+		finalLogLimit = agentLogLimit
```

Comment:
A debug log msg here would be useful to indicate we didn't accept the collector limit.

Reply:
Nice idea - added in commit [b4a4ce7](https://github.com/newrelic/newrelic-php-agent/pull/497/commits/b4a4ce7ed72d0e97810f741882ad1eca9a486ce6).